lib/inertia_rails/renderer.rb

```diff
@@ -64,24 +66,34 @@ def shared_data
     # Functionally, this permits using either string or symbol keys in the controller. Since the results
     # is cast to json, we should treat string/symbol keys as identical.
     def merge_props(shared_data, props)
-      shared_data.deep_symbolize_keys.send(@deep_merge ? :deep_merge : :merge, props.deep_symbolize_keys)
+      if @deep_merge
+        shared_data.deep_symbolize_keys.deep_merge!(props.deep_symbolize_keys)
+      else
+        shared_data.symbolize_keys.merge(props.symbolize_keys)
+      end
     end
 
     def computed_props
       _props = merge_props(shared_data, props).select do |key, prop|
         if rendering_partial_component?
-          key.in? partial_keys
+          key.in?(partial_keys) || prop.is_a?(AlwaysProp)
         else
-          !prop.is_a?(InertiaRails::Lazy)
+          !prop.is_a?(LazyProp)
         end
       end
 
-      deep_transform_values(
-        _props,
-        lambda do |prop|
-          prop.respond_to?(:call) ? controller.instance_exec(&prop) : prop
+      drop_partial_except_keys(_props) if rendering_partial_component?
```

Comment:
A possible nitpick, but could we return a new hash from this method instead of mutating `_props` directly?

Reply:
I'm trying not to create new objects when possible since we're working on the framework level and our implementation can affect app performance. Maybe I'm to protective here though. Need to come up with a good benchmarking example to test such things.